students/DennisLee/lesson04/mailroom.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+
+import os
+import pathlib
+
+# Initial donor list and the amounts they have donated
+donor_history = {
+            'Red Herring': [65820.5, 31126.37, 15000],
+            'Papa Smurf': [210.64, 1000],
+            'Pat Panda': [55324.4],
+            'Karl-Heinz Berthold': [3545.2, 10579.31],
+            'Mama Murphy': [156316.99, 8500.3, 12054.33],
+            'Daphne Dastardly': [82]
+        }
+
+def manage_donors():
+    """
+    Display the menu of choices for donor management.
+
+    :return:  None.
+    """
+    # create a dictionary of menu items, menu text, and menu caller functions
+    choices = {'1': ("Send a thank you", send_thank_you), 
+               '2': ("Create a report", create_a_report),
+               '3': ("Send letters to everyone", send_all_letters),
+               '4': ("Quit", exit_screen)}
+    
+    response = ''
+    while response != '4':  # Show menu forever until user exits
+        # Print the menu list (with numbered choices)
+        print()
+        for i in choices:
+            print(i, choices[i][0])
+
+        # Get the selection number
+        response = ''
+        while not response in choices:
+            response = input("Type your selection: ").strip()
+
+        choices[response][1]()  # Call helper function
+
+def exit_screen():
+    """
+    Simply print an exit message.
+
+    :return:  None.
+    """
+    print("\nExiting.\n")
+    return
+
+def send_thank_you():
+    """
+    Add new donations for new or existing donors, and send a thank-you
+    letter.
+
+    :return:  None.
+    """
+    # Get the donor name, show all donors, or quit
+    response = input(
+      "\nType the full donor name (or 'list' to show all donors, or 'quit'): "
+      ).strip()
+
+    if response.lower() in ('', 'quit'):
+        exit_screen()
+        return
+
+    elif response.lower() == 'list':
+        print("\nLIST OF DONORS:")
+        for donor in donor_history:
+            print(donor)
```

Comment:
We'll be adding unit tests to mailroom in a future assignment, so you might consider factoring the donor list generation out into its own function that returns the list and calling that here to make it more testable.